### DISCUSSION TITLE
streamlink twitch gui authethification token

### DISCUSSION THREAD

**Discussion by Mixalisfr** [streamlink/streamlink-twitch-gui#846] [General]
helloo i have a small problem signing with my account when i click sign in my browser(chrome redirects me to a bank page so i did a work around throught authenthification token thats dodnt seems to work eather i can watch streams but not with my account only as guest throught sreamlink twitch gui it tells me authethification failed try again

**Comment by bastimeyer** [comment]
When you click the "Sign in" button in Streamlink Twitch GUI, it first launches a local web server which listens on port 65432 and then it opens the following URL in your system's default web browser where you have to authorize the app with your Twitch account:
https://id.twitch.tv/oauth2/authorize?response_type=token&client_id=phiay4sq36lfv9zu7cbqwz2ndnesfd8&redirect_uri=http%3A%2F%2Flocalhost%3A65432%2Fredirect&scope=user_read+user_blocks_read+user_blocks_edit+user_follows_edit+user_subscriptions+chat_login&force_verify=true
On successful authentication, Twitch will redirect you to `http://localhost:65432` where the application can read the response from.

There is no "redirection to a blank page". It is possible that you have misconfigured your system and the webserver is not allowed to listen on port 65432, so the auth response can't get caught, but this does only happen after submitting the auth form on Twitch's website, so what you're saying doesn't make any sense. However if this is what you're trying to say, then you can still copy the auth token from the address bar of your web browser (on the `http://localhost:65432` page) and paste it into the Twitch GUI.

Authentication tokens are tokens for Twitch's kraken v5 API, which is only used by third party applications (and will be shut down at the end of February 2022 - the replace for that is the new helix API which hasn't been implemented here just yet). You can't use any other authentication token like for example one generated on Twitch's website, as these are used by different APIs. You'll also have to have the required authentication scope parameters registered on the token you're trying to use, which is described in the app if you click the "Show access token" button. This feature is not meant for users who don't know what they're doing.

**Reply by Mixalisfr** [reply]
i installed apache i was abled to login so i was able to sign in with the following steps 1) installing apache 2) login into twitch 3) authorize the twitch account from your link but fro the pop up chromium from the streamlink app

**Reply by bastimeyer** [reply]
> i installed apache

lol, no... The application is hosting a webserver on its own and shuts it off once the authentication succeeds or fails. This has literally nothing to do with anything.

> authorize the twitch account from your link but fro the pop up chromium from the streamlink app

There is no "Chromium popup". Early versions from like four years ago or even older were not using the system's web browser for authenticating and used an NW.js/Chromium popup, but I doubt that you're using this.

What you are writing is very confusing, once again. But sInce you managed to log in, I won't ask any further.
